opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1029,12 +1027,15 @@ namespace Opm {
                 const unsigned cellIdx = elemCtx.globalSpaceIndex(/*spaceIdx=*/0, /*timeIdx=*/0);
                 const auto& intQuants = elemCtx.intensiveQuantities(/*spaceIdx=*/0, /*timeIdx=*/0);
                 const auto& fs = intQuants.fluidState();
+                const double pv =
+                    ebosSimulator_.model().dofTotalVolume(cellIdx)
+                    * intQuants.porosity().value();
```

Comment:
Please add comment explaining why this is correct. Important points:
 - the intQuants.porosity() can be outside [0,1] due to pore volume manipulation
 - while the problem.porosity() would also account for this manipulation, only the intQuants version includes the pressure-dependent porosity multiplier